File: Java/EjercicioHerencia/src/ejercicioherencia/Empleado.java
package ejercicioherencia;

class Empleado extends Persona {
    
    //atributo de clase
    static private int contadorEmpleados;
    
    //atributo de instancia
    private int idEmpleado;
    private float sueldo;
    
    //constructor
    public Empleado(String nombre, String apellido, int edad, float sueldo) {
        super(nombre, apellido, edad);
        this.idEmpleado = ++Empleado.contadorEmpleados;
        this.sueldo = sueldo;
    }
    
    //getters & setters
    public int getIdEmpleado() {
        return this.idEmpleado;
    }

    public float getSueldo() {
        return this.sueldo;
    }

    public void setSueldo(float sueldo) {
        this.sueldo = sueldo;
    }
    
    //métodos
    @Override
    public String toString() {  
        return super.toString() + " " 
              + this.getIdEmpleado() + " " 
              + this.getSueldo();
    }
}

File: UJS2021-Udemy/Sección11/Empleado.js
class Empleado extends Persona {

    static contadorEmpleados = 0;
    
    constructor(nombre, apellido, edad, sueldo) {
        super(nombre, apellido, edad);
        this.sueldo = sueldo;
        this.idEmpleado = ++Empleado.contadorEmpleados;
    }

    //getters
    get getIdEmpleado() {
        return this.idEmpleado;
    }

    get getSueldo() {
        return this.sueldo;
    }
    
    //setters
    /**
     * @param {Number} sueldo
     */
    set setSueldo(sueldo) {
        this.sueldo = sueldo;
    }

    //Overriding
    toString() {
        return `${super.toString()} ${this.idEmpleado} ${this.sueldo}`;
        //return super.toString() + ' ' + this.idEmpleado + ' ' + this.sueldo;
        //return 'id=' + this.idEmpleado +': '+ this.nombre +' '+ this.apellido +', '+ this.edad + ' años' + ', ' + this.sueldo;
    }
}